tests/integration/api_v2/rpc/pendingTransactions.test.js

```diff
@@ -0,0 +1,204 @@
+			const response = await getTransactions({
+				includePending: true,
+				moduleAssetName: refTransaction.moduleAssetName,
+			});
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result.data).toBeInstanceOf(Array);
+			expect(result.data.length).toBeGreaterThanOrEqual(1);
+			expect(result.data.length).toBeLessThanOrEqual(10);
+			expect(response.result).toMap(resultEnvelopeSchema);
+			expect(result.data[0]).toMap(pendingTransactionSchemaVersion5);
+			expect(result.data[0]).toEqual(
+				expect.objectContaining({
+					isPending: true,
+				}),
+			);
+			expect(result.meta).toMap(metaSchema);
+		});
+
+		it('known transaction senderAddress -> ok', async () => {
+			const response = await getTransactions({
+				includePending: true,
+				senderAddress: refTransaction.sender.address,
+			});
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result.data).toBeInstanceOf(Array);
+			expect(result.data.length).toBeGreaterThanOrEqual(1);
+			expect(result.data.length).toBeLessThanOrEqual(10);
+			expect(response.result).toMap(resultEnvelopeSchema);
+			expect(result.data[0]).toMap(pendingTransactionSchemaVersion5);
+			expect(result.data[0]).toEqual(
+				expect.objectContaining({
+					isPending: true,
+				}),
+			);
+			expect(result.meta).toMap(metaSchema);
+		});
+
+		it('known transaction senderUsername -> ok', async () => {
+			const response = await getTransactions({
+				includePending: true,
+				senderUsername: refTransaction.sender.username,
+			});
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result.data).toBeInstanceOf(Array);
+			expect(result.data.length).toBeGreaterThanOrEqual(1);
+			expect(result.data.length).toBeLessThanOrEqual(10);
+			expect(response.result).toMap(resultEnvelopeSchema);
+			expect(result.data[0]).toMap(pendingTransactionSchemaVersion5);
+			expect(result.data[0]).toEqual(
+				expect.objectContaining({
+					isPending: true,
+				}),
+			);
+			expect(result.meta).toMap(metaSchema);
+		});
+
+		it('known transaction senderPublicKey -> ok', async () => {
+			const response = await getTransactions({
+				includePending: true,
+				senderPublicKey: refTransaction.sender.publicKey,
+			});
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result.data).toBeInstanceOf(Array);
+			expect(result.data.length).toBeGreaterThanOrEqual(1);
+			expect(result.data.length).toBeLessThanOrEqual(10);
+			expect(response.result).toMap(resultEnvelopeSchema);
+			expect(result.data[0]).toMap(pendingTransactionSchemaVersion5);
+			expect(result.data[0]).toEqual(
+				expect.objectContaining({
+					isPending: true,
+				}),
+			);
+			expect(result.meta).toMap(metaSchema);
+		});
+
+		it('empty transaction id -> ok', async () => {
+			const response = await getTransactions({
+				includePending: true,
+				transactionId: '',
```

Comment:
@MichalTuleja Should we be allowing users to send empty `transactionId` queries? Should we not add a non-empty validation here?

Reply:
I think this thing we can consider as an improvement and handle it at the gateway level because right now, we support the same (allow users to send queries with empty param) for other endpoints too.

Reply:
@sameersubudhi The empty params are supported and equivalent to the default value - as the param wasn't set at all. If that behaviour is preserved here, all is good 👍